parser_library/src/processing/statement_analyzers/hit_count_analyzer.cpp

```diff
@@ -0,0 +1,192 @@
+        if (resolved_stmt->opcode_ref().value != context::id_storage::well_known::MACRO)
+            return false;
+
+        ++nest_level;
+        return true;
+    };
+
+    static constexpr auto range_adj = [](range r) {
+        r.end.column = 72;
+        return r;
+    };
+
+    switch (m_stmt_type)
+    {
+        case hlasm_plugin::parser_library::processing::hit_count_analyzer::statement_type::REGULAR:
+            if (macro_init_incrementer(res_stmt))
+            {
+                m_stmt_type = statement_type::MACRO_INIT;
+                m_last_macro_init_r = res_stmt->stmt_range_ref();
+            }
+            break;
+
+        case hlasm_plugin::parser_library::processing::hit_count_analyzer::statement_type::MACRO_INIT: {
+            m_stmt_type = statement_type::MACRO_NAME;
+
+            assert(m_ctx.processing_stack().frame().resource_loc);
+
+            m_macro_header_definitions.try_emplace(res_stmt->opcode_ref().value.to_string(),
+                macro_header_definition_details { *m_ctx.processing_stack().frame().resource_loc,
+                    range_adj(std::move(m_last_macro_init_r)),
+                    range_adj(res_stmt->stmt_range_ref()) });
+            break;
+        }
+
+        case hlasm_plugin::parser_library::processing::hit_count_analyzer::statement_type::MACRO_NAME:
+            m_stmt_type = statement_type::MACRO_BODY;
+            macro_init_incrementer(res_stmt);
+            break;
+
+        case hlasm_plugin::parser_library::processing::hit_count_analyzer::statement_type::MACRO_BODY:
+            if (res_stmt->opcode_ref().value == context::id_storage::well_known::MEND)
+            {
+                if (--m_macro_nest_level == 0)
+                    return std::exchange(m_stmt_type, statement_type::REGULAR);
+            }
+            else
+                macro_init_incrementer(res_stmt);
+            break;
+
+        default:
+            break;
+    }
+
+    return m_stmt_type;
+}
+
+void hit_count_analyzer::analyze(
+    const context::hlasm_statement& statement, statement_provider_kind prov_kind, processing_kind proc_kind, bool)
+{
+    bool can_be_external_macro = false;
+
+    switch (get_stmt_type(statement))
+    {
+        using enum processing::hit_count_analyzer::statement_type;
+        case REGULAR:
+            break;
+
+        case MACRO_BODY:
+            can_be_external_macro = true;
+            break;
+
+        default:
+            return;
+    }
+
+    auto stmt_range = get_range(&statement, prov_kind, proc_kind, m_ctx);
+    if (!stmt_range)
+        return;
+
+    if (auto macro_invo = m_ctx.this_macro(); macro_invo)
+        emplace_macro_header_definitions(macro_invo->id);
+
+    assert(m_ctx.processing_stack().frame().resource_loc);
+
+    auto& count = emplace_hit_count(
+        m_hit_counts, *m_ctx.processing_stack().frame().resource_loc, std::move(*stmt_range), can_be_external_macro);
```

Comment:
There is a simpler/faster processing_stack_top() that only returns the top frame.